pkgs/leak_tracker/lib/src/leak_tracking/primitives/_test_helper_detector.dart

```diff
@@ -53,3 +49,37 @@ bool isCreatedByTestHelper(
   }
   return false;
 }
+
+/// Postponed detector of test helpers.
+///
+/// It is used to detect if the leak was created by a test helper,
+/// only if the leak is detected.
+///
+/// It is needed because the detection is a heavy operation
+/// and should not be done for every tracked object.
+class CreationChecker {
+  StackTrace? _creationStack;
+  List<RegExp>? _exceptions;
+
+  /// True, if the leak was created by a test helper.
+  ///
+  /// This value is cached. The first calculation of the value
+  /// is performance heavy.
+  late final bool createdByTestHelpers = () {
+    final result = isCreatedByTestHelper(
+      _creationStack!.toString(),
+      _exceptions!,
+    );
+    _creationStack = null;
+    _exceptions = null;
+    return result;
+  }();
+
+  /// Creates instance of [CreationChecker].
+  ///
+  /// Frames, that match [exceptions] will be ignored.
```

Comment:
nit: to make this more clear "Stack frames in [creationStack] that match any of [exceptions] will be ignored."